modules/programs/fd.nix

```diff
@@ -0,0 +1,79 @@
+{ config, lib, pkgs, ... }:
+with lib; {
+  meta.maintainers = [ maintainers.uncenter ];
+
+  options.programs.fd = {
+    enable = mkEnableOption
+      "fd, a simple, fast and user-friendly alternative to {command}`find`";
+
+    enableBashIntegration = mkEnableOption "Bash integration" // {
+      default = true;
+    };
+
+    enableZshIntegration = mkEnableOption "Zsh integration" // {
+      default = true;
+    };
+
+    enableFishIntegration = mkEnableOption "Fish integration" // {
+      default = true;
+    };
+
+    enableIonIntegration = mkEnableOption "Ion integration" // {
+      default = true;
+    };
+
+    enableNushellIntegration = mkEnableOption "Nushell integration" // {
+      default = true;
+    };
+
+    ignores = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ ".git/" "*.bak" ];
+      description = "List of paths that should be globally ignored.";
+    };
+
+    hidden = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Search hidden files and directories ({option}`--hidden` argument).
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = [ "--no-ignore" "--absolute-path" ];
+      description = ''
+        Extra command line options passed to fd.
+      '';
+    };
```

Comment:
Might be better to do something more like ripgrep
https://github.com/nix-community/home-manager/blob/fa8c16e2452bf092ac76f09ee1fb1e9f7d0796e7/modules/programs/ripgrep.nix#L17-L28

Reply:
I'm going with what the `eza` module did, by having one of the rather important options separate.